### PR TITLE
Allow client r30 data to be reported in metrics db

### PR DESCRIPTION
**Pull request by callahad**
This is a bit gnarly, but it works.

### CODE REVIEW

File: retention/r30.py [hunk 1]
```diff
@@ -257,9 +254,22 @@ def mysql_reporter(**kwargs):
     cursor = db.cursor()
 
     def report(date, r30, **client_r30):
```
Comment (anoadragon453) [inline retention/r30.py]: Could you add a docstring to this function describing what each argument is? (It took me a while to figure out what `client_r30` actually was - it appears to be a dictionary of `client_name: count`).

Also it's odd that we deconstruct `client_r30` when passing it to `report`... which just constructs it back into a dict. But outside of the scope of this PR I guess :slightly_smiling_face:

File: retention/r30.py [hunk 2]
```diff
@@ -257,9 +254,22 @@ def mysql_reporter(**kwargs):
...
+        values = [date, r30]
+
+        if client_r30:
+            # Only accept known clients per the constant above
+            client_r30 = {k: v for k, v in client_r30.items() if k in CLIENTS}
+
+            # Split the client data into two lists: one of names, one of values
+            extra_columns, extra_values = zip(*client_r30.items())
+            columns.extend(extra_columns)
+            values.extend(extra_values)
```
Comment (anoadragon453) [inline retention/r30.py]:
```suggestion
            client_names, client_count = zip(*client_r30.items())
            columns.extend(client_names)
            values.extend(client_count)
```

For clarity?

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
lgtm otherwise, I think

File: retention/r30.py (hunk 1 above)
Comment (callahad) [reply retention/r30.py]: Ah, yeah, that's a remnant from when I was only using the CSV DictWriter... the same signature just ended up getting carried over. Doesn't really make sense anymore. Fixing (+ docstring)

File: retention/r30.py (hunk 2 above)
Comment (anoadragon453) [reply retention/r30.py]: Yay/neigh to this?

**Review by anoadragon453 — APPROVED** [review]
TIL about bare `*` in function args forcing callers to specify argument names!

I think you might've missed one of the review comments, but I'm happy to go either way on it.

File: retention/r30.py (hunk 2 above)
Comment (callahad) [reply retention/r30.py]: Sure!

**Pull request merged**